src/task_types/fuzzer_tasks.rs

```diff
@@ -216,27 +217,24 @@ impl GenFuzzerService for GenFuzzerServiceImpl {
         let task_name = &params.task_name;
         let mut sub_tasks: Vec<EvgTask> = vec![];
         if params.is_multiversion() {
+            if params.multiversion_generate_tasks.is_empty() {
+               panic!("Multiversion task definition expected to have 'multiversion_generate_tasks' but did not: `{}`", params.task_name)
+            }
             event!(
                 Level::INFO,
                 task_name = task_name.as_str(),
                 "Generating multiversion fuzzer"
             );
-            for (old_version, version_combination) in
-                self.multiversion_service.multiversion_iter(&params.suite)?
-            {
-                let base_task_name =
-                    build_name(&params.task_name, &old_version, &version_combination);
-                let base_suite_name = build_name(&params.suite, &old_version, &version_combination);
-
+            for multiversion_task in &params.multiversion_generate_tasks {
                 sub_tasks.extend(
                     (0..params.num_tasks as usize)
                         .map(|i| {
                             build_fuzzer_sub_task(
-                                &base_task_name,
+                                &multiversion_task.suite_name,
```

Comment:
This makes it so that the execution task has the same name as the suite, which will make things more clear.

Have a failing execution task? Copy the string and search for it in the repo, you should immediately see the matrix suite file, the generated suite file, and the "initialize multiversion tasks" var name in `definitions.yml`.

Before these changes, you can search for the failing execution task string, but wont find any hits in the repo.

Reply:
I think this sounds like a helpful change, but can you put an example here? I just would like to see what this looks like.

Also, have you synced with someone from Server regarding this? I just want to make sure there won't be any confusion or cause any issues when this changes.

Reply:
Old generated config
```json
  {
      "name": "aggregation_multiversion_fuzzer_last_lts_1-linux",
      "commands": [
        {
          "func": "git get project no modules"
        },
        {
          "func": "add git tag"
        },
        {
          "func": "do setup"
        },
        {
          "func": "configure evergreen api credentials"
        },
        {
          "func": "do multiversion setup"
        },
        {
          "func": "setup jstestfuzz"
        },
        {
          "func": "run jstestfuzz",
          "vars": {
            "npm_command": "agg-fuzzer",
            "jstestfuzz_vars": "--numGeneratedFiles 5 "
          }
        },
        {
          "func": "run generated tests",
          "vars": {
            "timeout_secs": 1800,
            "suite": "generational_fuzzer_last_lts",
            "resmoke_jobs_max": 1,
            "resmoke_args": "--mongodSetParameters='{logComponentVerbosity: {command: 2}}'",
            "gen_task_config_location": "mongodb-mongo-master/abc123/generate_tasks/generated-config-321ab.tgz",
            "require_multiversion_setup": true,
            "task": "aggregation_multiversion_fuzzer",
            "multiversion_exclude_tags_version": "last_lts",
            "continue_on_failure": false,
            "should_shuffle": false
          }
        }
      ],
      "depends_on": [
        {
          "name": "version_gen"
        },
        {
          "name": "archive_dist_test_debug"
        }
      ]
    }
```
New generated config
```json
   {
      "name": "aggregation_multiversion_fuzzer_last_lts_1-linux",
      "commands": [
        {
          "func": "git get project no modules"
        },
        {
          "func": "add git tag"
        },
        {
          "func": "do setup"
        },
        {
          "func": "configure evergreen api credentials"
        },
        {
          "func": "do multiversion setup"
        },
        {
          "func": "setup jstestfuzz"
        },
        {
          "func": "run jstestfuzz",
          "vars": {
            "npm_command": "agg-fuzzer",
            "jstestfuzz_vars": "--numGeneratedFiles 5 "
          }
        },
        {
          "func": "run generated tests",
          "vars": {
            "resmoke_jobs_max": 1,
            "gen_task_config_location": "mongodb-mongo-master/abc123/generate_tasks/generated-config-321ab.tgz",
            "timeout_secs": 1800,
            "multiversion_exclude_tags_version": "last_lts",
            "suite": "aggregation_multiversion_fuzzer_last_lts",
            "should_shuffle": false,
            "task": "aggregation_multiversion_fuzzer",
            "resmoke_args": "--mongodSetParameters='{logComponentVerbosity: {command: 2}}'",
            "require_multiversion_setup": true,
            "continue_on_failure": false
          }
        }
      ],
      "depends_on": [
        {
          "name": "version_gen"
        },
        {
          "name": "archive_dist_test_debug"
        }
      ]
    }
```

If you look at this section, you'll see that in the new config, the task name is identical to the suite name. Therefore, when a dev sees that task failing in evergreen it is very easy to determine search and find the suite. In the old config you will see that the task name and suite name differ. For a dev to find the underlying suite that is running in the task, it requires a lot of digging.
```json
{
          "func": "run generated tests",
          "vars": {
            ...
            "suite": "...",
            ...
          }
        }
```

Regarding syncing with SERVER, there should be no noticable changes on their side of things as a result of these changes. But, a new requirement of this API is that each `multiversion_task.suite_name` must be unique (this is because we now use the `multiversion_task.suite_name` as the task name & evergreen requires unique task names). More context [here](https://github.com/10gen/mongo/pull/11789#discussion_r1172939305). I am syncing with a few people on SERVER on that.